File: customview/src/main/java/com/example/customview/tagview/TagAdapter.java
package com.example.customview.tagview;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.example.customview.R;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by test on 2017/4/14.
 * <p>
 * 评论页面的适配器
 */

public class TagAdapter extends RecyclerView.Adapter<TagAdapter.ViewHolder> {

    private List<TagBean> tagList;

    private boolean isSelected = false;

    private List<TagBean> selectList;

    public TagAdapter(List<TagBean> tagList) {
        this.tagList = tagList;
        selectList = new ArrayList<>();
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.tag_layout, parent, false);
        ViewHolder vh = new ViewHolder(view);
        return vh;
    }

    @Override
    public void onBindViewHolder(final ViewHolder holder, final int position) {
        holder.mTextView.setText(tagList.get(position).getTag_name());
        holder.itemView.setTag(tagList.get(position));
        holder.mTextView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                isSelected = !holder.mTextView.isSelected();
                if (isSelected) {
                    holder.mTextView.setSelected(true);
                    holder.mTextView.setBackgroundResource(R.drawable.tag_checked_bg);
                    selectList.add(tagList.get(position));
                } else {
                    holder.mTextView.setSelected(false);
                    holder.mTextView.setBackgroundResource(R.drawable.tag_normal_bg);
                    selectList.remove(tagList.get(position));
                }
            }
        });

    }

    @Override
    public int getItemCount() {
        return tagList.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        public TextView mTextView;

        public ViewHolder(View view) {
            super(view);
            mTextView = view.findViewById(R.id.tag_tv);
        }
    }

    public List<TagBean> getSelectData() {
        return selectList;
    }
}

File: customview/src/main/java/com/example/customview/tagview/TagAdapter.kt
package com.example.customview.tagview

import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.customview.R
import com.example.customview.databinding.TagLayoutBinding

/**
 * Created by test on 2017/4/14.
 *
 * 评论页面的适配器
 */
open class TagAdapter(private var tagList: List<TagBean>) :
    RecyclerView.Adapter<TagAdapter.ViewHolder>() {
    private var isSelected = false
    private val selectList = ArrayList<TagBean>()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val binding = TagLayoutBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return ViewHolder(binding)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.mTextView.text = tagList[position].tag_name
        holder.itemView.tag = tagList[position]
        holder.mTextView.setOnClickListener {
            isSelected = !holder.mTextView.isSelected
            if (isSelected) {
                holder.mTextView.isSelected = true
                holder.mTextView.setBackgroundResource(R.drawable.tag_checked_bg)
                selectList.add(tagList[position])
            } else {
                holder.mTextView.isSelected = false
                holder.mTextView.setBackgroundResource(R.drawable.tag_normal_bg)
                selectList.remove(tagList[position])
            }
        }
    }

    override fun getItemCount(): Int {
        return tagList.size
    }

    inner class ViewHolder(binding: TagLayoutBinding) : RecyclerView.ViewHolder(binding.root) {
        @JvmField
        var mTextView: TextView = binding.tagTv
    }

    fun getSelectData(): List<TagBean> {
        return selectList
    }
}
